atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TransactionKeyValueService.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.api;
+
+import com.google.common.collect.Multimap;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.errorprone.annotations.MustBeClosed;
+import com.palantir.atlasdb.keyvalue.api.BatchColumnRangeSelection;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.ColumnRangeSelection;
+import com.palantir.atlasdb.keyvalue.api.ColumnSelection;
+import com.palantir.atlasdb.keyvalue.api.KeyAlreadyExistsException;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RowColumnRangeIterator;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.common.base.ClosableIterator;
+import com.palantir.processors.AutoDelegate;
+import com.palantir.util.paging.TokenBackedBasicResultsPage;
+import java.util.Map;
+
+/**
+ * Key-Value API to be used with transactional tables.
+ *
+ * Close cousin of {@link com.palantir.atlasdb.keyvalue.api.KeyValueService}
+ * but severely restricts the API available to transactions.
+ */
+@AutoDelegate
+public interface TransactionKeyValueService {
```

Comment:
I don't know if "Transaction" is the naming I want. Traditionally this would probably be something long the lines of "MultiVersionConcurrencyStore" but that doesn't roll off the tongue exactly either.